### PR TITLE
Disk Chaos API

### PR DESCRIPTION
**Pull request by Andrewmatilde**
### What problem does this PR solve?
Add chaos api of `disk attack` in chaosd to Chaos-Mesh.

 
Related issue #2153 

### What's changed and how it works?
Add a new chaos api.

### Related changes

- [x] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test

### CODE REVIEW

**Comment by Andrewmatilde** [comment]
PTAL @STRRL , @WangXiangUSTC

File: api/v1alpha1/diskchaos_types.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	Status DiskChaosStatus `json:"status"`
+}
+
+var _ InnerObjectWithSelector = (*DiskChaos)(nil)
+var _ InnerObject = (*DiskChaos)(nil)
+
+type DiskChaosAction string
+
+const (
+	Fill   DiskChaosAction = "fill"
```
Comment (WangXiangUSTC) [inline api/v1alpha1/diskchaos_types.go:45]: how about renaming it, so it has the same style as others?

File: api/v1alpha1/diskchaos_types.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	ContainerSelector `json:",inline"`
+
+	// +kubebuilder:validation:Enum=fill;write;read
+	Action DiskChaosAction `json:"action" webhook:"DiskAction"`
+
+	Path    string `json:"path,omitempty"`
+	Size    string `json:"size,omitempty"`
+	Percent string `json:"percent,omitempty"`
+
+	SpaceLockSize string `json:"space_lock_size,omitempty"`
```
Comment (WangXiangUSTC) [inline api/v1alpha1/diskchaos_types.go:60]: This config is not provided in Chaosd, what is this config used to do? It's better to add some comments.

File: api/v1alpha1/diskchaos_types.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+	FillByFAllocate bool `json:"fill_by_fallocate,omitempty"`
+
+	ProcessNum    uint8 `json:"process_num,omitempty" webhook:"ProcessNum"`
+	LoopExecution bool  `json:"loop_execution,omitempty"`
```
Comment (WangXiangUSTC) [inline api/v1alpha1/diskchaos_types.go:65]: ditto

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: api/v1alpha1/diskchaos_types.go (hunk 1 above)
Comment (Andrewmatilde) [reply api/v1alpha1/diskchaos_types.go:45]: I agree with you.

Comment (Andrewmatilde) [reply api/v1alpha1/diskchaos_types.go:45]: Fixed.

File: api/v1alpha1/diskchaos_types.go (hunk 2 above)
Comment (Andrewmatilde) [reply api/v1alpha1/diskchaos_types.go:62]: Fixed.

File: api/v1alpha1/diskchaos_types.go (hunk 3 above)
Comment (Andrewmatilde) [reply api/v1alpha1/diskchaos_types.go]: This config will not provided soon , so I add `not implement` here.

**Comment by zamazan4ik** [comment]
What is the current status for the PR?

**Comment by Andrewmatilde** [comment]
> What is the current status for the PR?

Blocking on some related chaosctl problem. I will fix it later with @Hexilee .

**Comment by cwen0** [comment]
> > What is the current status for the PR?
> 
> Blocking on some related chaosctl problem. I will fix it later with @Hexilee .

Do you have any updates about this? Is this pr ready for review? @Andrewmatilde

**Comment by Andrewmatilde** [comment]
> Do you have any updates about this? Is this pr ready for review? @Andrewmatilde

Yes , but I need to review it by own at once.

**Pull request closed without merging**